include/stellar/stellar_launcher.hpp

```diff
@@ -0,0 +1,136 @@
+    }
+
+    static StellarComputeStatistics
+    search_and_verify(
+        StellarDatabaseSegment<TAlphabet> const databaseSegment,
+        TId const & databaseID,
+        QueryIDMap<TAlphabet> const & queryIDMap,
+        bool const databaseStrand,
+        StellarOptions & localOptions, // localOptions.compactThresh is out-param
+        StellarSwiftPattern<TAlphabet> & localSwiftPattern,
+        stellar::stellar_kernel_runtime & strand_runtime,
+        StringSet<QueryMatches<StellarMatch<String<TAlphabet> const, TId> > > & localMatches
+    )
+    {
+        using TSequence = String<TAlphabet>;
+
+        auto getQueryMatches = [&](auto const & pattern) -> QueryMatches<StellarMatch<TSequence const, TId> > &
+        {
+            // Note: Index is normally build over all queries [query0, query1, query2, ...],
+            // but in LocalQueryPrefilter case it can just be build over [query0].
+            // We need to translate that position to a "record" ID
+            //!TODO: this shouldn't be necessary
+            // each Stellar instance should be given a set of bin query, all of which should be indexed
+            size_t const queryRecordID = queryIDMap.recordID(pattern);
+            return value(localMatches, queryRecordID);
+        };
+
+        auto isPatternDisabled = [&](StellarSwiftPattern<TAlphabet> & pattern) -> bool {
+            QueryMatches<StellarMatch<TSequence const, TId> > & queryMatches = getQueryMatches(pattern);
+            return queryMatches.disabled;
+        };
+
+        auto onAlignmentResult = [&](auto & alignment) -> bool {
+            QueryMatches<StellarMatch<TSequence const, TId> > & queryMatches = getQueryMatches(localSwiftPattern);
+
+            StellarMatch<TSequence const, TId> match(alignment, databaseID, databaseStrand);
+            length(match);  // DEBUG: Contains assertion on clipping.
+
+            // success
+            return _insertMatch(
+                queryMatches,
+                match,
+                localOptions.minLength,
+                localOptions.disableThresh,
+                // compactThresh is basically an output-parameter; will be updated in kernel and propagated back
+                // outside of this function, the reason why StellarOptions can't be passed as const to this function.
+                // TODO: We might want to make this tied to the QueryMatches itself, as it should know then to consolidate
+                // the matches
+                localOptions.compactThresh,
+                localOptions.numMatches
+            );
+        };
+
+        // finder
+        StellarSwiftFinder<TAlphabet> swiftFinder(databaseSegment.asInfixSegment(), localOptions.minRepeatLength, localOptions.maxRepeatPeriod);
+
+        StellarComputeStatistics statistics = _verificationMethodVisit(
+            localOptions.verificationMethod,
+            [&](auto tag) -> StellarComputeStatistics
+            {
+                using TTag = decltype(tag);
```

Comment:
In c++20 we can now do:
```suggestion
            [&]<typename TTag>(TTag tag) -> StellarComputeStatistics
            {
```